src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,66 @@
+package duke.task;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Represents the deadline type task that has a deadline date and an optional parsedDate
+ * if provided date string is parsable
+ */
+public class Deadline extends ListItem {
+    private final String date;
+    private LocalDate parsedDate;
+
+    public Deadline(String task, String inputDate) {
+        super(task);
+        this.date = inputDate;
+        this.parsedDate = parseDate(inputDate);
+    }
+
+    /**
+     * the overloaded constructor that allows taking the status of the task
+     * @param task the task name which is passed to parent to store
+     * @param inputDate date entered as the deadline
+     * @param isDone the status of the task
+     */
+    public Deadline(String task, String inputDate, boolean isDone) {
+        super(task, isDone);
+        this.date = inputDate;
+        this.parsedDate = parseDate(inputDate);
+    }
+
+    /**
+     * Changes the task's status to be done
+     * @return the task to replace the old one in the list or to be used later
+     */
+    @Override
+    public ListItem markAsDone() {
+        return new Deadline(super.getTask(), (parsedDate == null ? this.date : parsedDate.format(DateTimeFormatter.ofPattern("MMM d yyyy"))), true);
+    }
+
+    @Override
+    public String toString() {
+        return "[D]" + (super.getDone() == true ? "[X] " : "[ ] ") + super.getTask() + " (by: " + (parsedDate == null ? this.date : parsedDate.format(DateTimeFormatter.ofPattern("MMM d yyyy"))) + ")";
```

Comment:
This line is pretty long. I think you can split this into more lines. I like the usage of one-liner if-else, though :D

Reply:
Agree! :)